interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -287,6 +354,36 @@ func (pdb *PromptDB) AddOrMerge(metadata *prompting.Metadata, path string, reque
 			ids: make(map[prompting.IDType]int),
 		}
 		userEntry = pdb.perUser[metadata.User]
+		userEntry.expirationTimer = time.AfterFunc(initialTimeout, func() {
+			pdb.mutex.Lock()
+			if pdb.maxIDMmap.IsClosed() {
+				return
+			}
+			expiredPrompts := userEntry.prompts
+			// Clear all outstanding prompts for the user
+			userEntry.prompts = nil
+			userEntry.ids = make(map[prompting.IDType]int)
+			// Restart expiration timer while holding the lock, so we don't
+			// overwrite a newly-set activity timeout with an initial timeout.
+			// With the lock held, no activity can occur, so no activity timeout
+			// can be set.
+			if userEntry.expirationTimer.Reset(initialTimeout) {
+				// Timer was active again, suggesting that some activity caused
+				// the timer to be reset at some point between the timer firing
+				// and the lock being released and subsequently acquired by this
+				// function. So reset the timer to activityTimeout, and do not
+				// purge prompts.
```

Comment:
hm but the prompts have been purged just a few lines above, haven't they?

Reply:
They were, my mistake. Fixed now